flang/include/flang/Lower/Runtime.h

```diff
@@ -73,6 +74,26 @@ void genDateAndTime(FirOpBuilder &, mlir::Location,
                     llvm::Optional<fir::CharBoxValue> date,
                     llvm::Optional<fir::CharBoxValue> time,
                     llvm::Optional<fir::CharBoxValue> zone);
+void genAllocatableInitIntrinsic(FirOpBuilder &, mlir::Location,
+                                 mlir::Value boxAddress,
+                                 mlir::Value typeCategory, mlir::Value kind,
+                                 mlir::Value rank, mlir::Value corank);
+
+void genAllocatableSetBounds(FirOpBuilder &builder, mlir::Location loc,
+                             mlir::Value boxAddress, mlir::Value dimIndex,
+                             mlir::Value lowerBoud, mlir::Value upperBound);
+
+mlir::Value genAllocatableAllocate(FirOpBuilder &builder, mlir::Location loc,
+                                   mlir::Value boxAddress, mlir::Value hasStat,
+                                   mlir::Value errMsgBox,
+                                   mlir::Value sourceFile,
+                                   mlir::Value sourceLine);
+
+mlir::Value genAllocatableDeallocate(FirOpBuilder &builder, mlir::Location loc,
+                                     mlir::Value boxAddress,
+                                     mlir::Value hasStat, mlir::Value errMsgBox,
+                                     mlir::Value sourceFile,
+                                     mlir::Value sourceLine);
```

Comment:
You' right, I had put them here to avoid copy pasting `genRuntimeFunction`. I moved `genRuntimeFunction` in `RTBuilder.h` (and modified it a bit, the indirection through the tuple was making it hard to share this logic, see last commit). With that I moved that to Allocatable.cpp as suggested.